Zend/zend_portability.h

```diff
@@ -864,4 +865,21 @@ static zend_always_inline uint64_t ZEND_BYTES_SWAP64(uint64_t u)
 # define ZEND_OPCACHE_SHM_REATTACHMENT 1
 #endif
 
+#ifndef CHAR_BITS
+# define CHAR_BITS 8
+#endif
+
+#ifndef UINTPTR_WIDTH
+# define UINTPTR_WIDTH (CHAR_BITS * sizeof(uintptr_t))
+#endif
+
+#if ((defined(__linux__) && defined(__x86_64__)) \
+		|| (defined(ZEND_WIN32) && defined(_M_AMD64))) \
+		&& !defined(NO_ZEND_HIGH_HALF_KERNEL)
+/* The kernel reserves the higher part of the address space for itself.
+ * Therefore, we can assume that the higher bit of user space addresses is
+ * never set. */
+# define ZEND_HIGH_HALF_KERNEL
```

Comment:
So I didn't like this higher half stuff because it adds more complexity, but I'm glad to see it doesn't make an improvement so this can be simplified